src/openklant2/tests/helpers.py

```diff
@@ -0,0 +1,177 @@
+            )
+            raise
+
+    def _manage_py(
+        self,
+        *args: str,
+        input: str | None = None,
+    ):
+        self._docker_compose(
+            "run",
+            "--rm",
+            self._django_service_name,
+            "python",
+            "src/manage.py",
+            *args,
+            input=input,
+        )
+
+    def _service_teardown(self):
+        self._docker_compose("kill", check=False)
+        self._docker_compose("down", "-v")
+        self._docker_compose("rm", "-f")
+
+    def _service_init(self):
+        self._docker_compose("up", "-d")
+        self._wait_for_response()
+        self._manage_py("migrate")
+
+    def _reset_db_state(self):
+        self._manage_py("flush", "--no-input")
+        self._load_fixture_from_json_string(self._generate_token_fixture())
+
+    def _load_fixture_from_json_string(self, fixture: str):
+        self._manage_py(
+            "loaddata",
+            "--format",
+            "json",
+            "-",  # i.e. stdin
+            input=fixture,
+        )
+
+    def _generate_token_fixture(self):
+        return json.dumps(
+            [
+                {
+                    "model": "token.tokenauth",
+                    "pk": 1,
+                    "fields": {
+                        "token": self._api_token,
+                        "contact_person": "Boaty McBoatface",
+                        "email": "boaty@mcboatface.com",
+                        "organization": "",
+                        "last_modified": "2024-08-22T07:43:21.837Z",
+                        "created": "2024-08-22T07:43:21.837Z",
+                        "application": "",
+                        "administration": "",
+                    },
+                }
+            ]
+        )
+
+    def _wait_for_response(self, interval=0.5, max_wait=60):
+        start_time = time.time()
+        while True:
+            try:
+                response = requests.get(self._service_url)
+                return response
+            except requests.RequestException:
+                logger.debug("Exception while checking for liveness", exc_info=True)
+                elapsed_time = time.time() - start_time
+                if elapsed_time >= max_wait:
+                    logger.info("Max wait time exceeded.")
+                    raise RuntimeError(
+                        f"Maximum wait for service to be healthy exceeded: {elapsed_time} > {max_wait}"
+                    )
+
+                time.sleep(interval)
+
+    def __enter__(self):
+        if self._in_server_context:
+            raise RuntimeError(
+                "You cannot have multiple server contexts active at the same time"
+            )
+
+        self._in_server_context = True
+        self._service_teardown()
+        self._service_init()
+
+        return self
+
+    @property
+    def api_root(self):
+        return f"{self._service_url}{self._api_base_path}"
```

Comment:
Could use `urljoin` if we want to make these configurable (otherwise doesn't matter).